packages/components/react/src/Panels/WorkspacePanel.tsx

```diff
@@ -71,157 +106,147 @@ export function WorkspacePanel({ tutorialStore, theme }: Props) {
     } else {
       tutorialStore.reset();
     }
-  }, [tutorialStore.ref]);
+  }
 
   useEffect(() => {
-    const lesson = tutorialStore.lesson!;
-
-    const unsubscribe = tutorialStore.lessonFullyLoaded.subscribe((loaded) => {
-      if (loaded && lesson.data.autoReload) {
-        /**
-         * @todo This causes some race with the preview where the iframe can show the "wrong" page.
-         * I think the reason is that when the ports are different then we render new frames which
-         * races against the reload which will internally reset the `src` attribute.
-         */
-        // previewRef.current?.reload();
-      }
-    });
```

Comment:
Btw, I think we should re-introduce this logic. It used to be broken because previews were loosing their state between lessons. Because now they are not re-parented anymore, I think this logic should work as expected.

Maybe before removing it, we should have a PR that add it back and do some checks that it works with a simple node http server that does not do livereload?

Reply:
I'll check this! So what kind of setup is required, and what used to be the erroneous state that was seen here? Add two lessons with Node servers on each and make sure correct content is shown when navigating between the lessons?

Reply:
Yep, this works! Though there's a bug in iframe selector here. The `preview-panel` never contains any `iframe` elements as children - they are some how mirrored from the `#previews-container` there 🤔 

https://github.com/stackblitz/tutorialkit/blob/98ef05b828ff8f3ab45a49e62bf1a4b79e65acfc/packages/react/src/Panels/PreviewPanel.tsx#L52-L59

Reply:
Continues at https://github.com/stackblitz/tutorialkit/pull/303